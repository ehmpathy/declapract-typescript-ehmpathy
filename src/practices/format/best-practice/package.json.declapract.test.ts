import { contents } from './package.json.declapract';

describe('package.json', () => {
  it('it should include terraform formatting when terraform practice is in use', async () => {
    const declaredBestPracticeContents = await contents({
      projectPractices: ['terraform'],
    } as any);
    expect(declaredBestPracticeContents).toContain('fix:format:terraform');
    expect(declaredBestPracticeContents).toMatchSnapshot();
  });
  it('should include not config mock when project does not use config practice', async () => {
    const declaredBestPracticeContents = await contents({
      projectPractices: [],
    } as any);
    expect(declaredBestPracticeContents).not.toContain('fix:format:terraform');
    expect(declaredBestPracticeContents).toMatchSnapshot();
  });
});
