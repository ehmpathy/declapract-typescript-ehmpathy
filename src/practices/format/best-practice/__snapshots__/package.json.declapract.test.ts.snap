// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`package.json it should include terraform formatting when terraform practice is in use 1`] = `
"{
  "scripts": {
    "fix:format": "npm run fix:format:prettier && npm run fix:format:terraform",
    "test:format": "npm run test:format:prettier && npm run test:format:terraform"
  }
}"
`;

exports[`package.json should include not config mock when project does not use config practice 1`] = `
"{
  "scripts": {
    "fix:format": "npm run fix:format:prettier",
    "test:format": "npm run test:format:prettier"
  }
}"
`;
